test_enviroment/os_interface.py

```diff
@@ -0,0 +1,204 @@
+        else:
+            auth_url = 'https://{0}:5000/v2.0/'.format(self.controller_ip)
+            path_to_cert = PATH_TO_CERT
+
+        logger.debug('Auth URL is {0}'.format(auth_url))
+        self.nova = novaclient(username=user,
+                               api_key=password,
+                               project_id=tenant,
+                               auth_url=auth_url,
+                               cacert=path_to_cert)
+
+        self.cinder = cinderclient.Client(1, user, password,
+                                          tenant, auth_url,
+                                          cacert=path_to_cert)
+
+        self.neutron = neutronclient.Client(username=user,
+                                            password=password,
+                                            tenant_name=tenant,
+                                            auth_url=auth_url,
+                                            ca_cert=path_to_cert)
+
+        self.keystone = self._get_keystoneclient(username=user,
+                                                 password=password,
+                                                 tenant_name=tenant,
+                                                 auth_url=auth_url,
+                                                 ca_cert=path_to_cert)
+
+        token = self.keystone.auth_token
+        logger.debug('Token is {0}'.format(token))
+        glance_endpoint = self.keystone.service_catalog.url_for(
+            service_type='image', endpoint_type='publicURL')
+        logger.debug('Glance endpoind is {0}'.format(glance_endpoint))
+
+        self.glance = glanceclient(endpoint=glance_endpoint,
+                                   token=token,
+                                   cacert=path_to_cert)
+
+    def _get_keystoneclient(self, username, password, tenant_name, auth_url,
+                            retries=3, ca_cert=None):
+        keystone = None
+        for i in range(retries):
+            try:
+                if ca_cert:
+                    keystone = keystoneclient(username=username,
+                                              password=password,
+                                              tenant_name=tenant_name,
+                                              auth_url=auth_url,
+                                              cacert=ca_cert)
+
+                else:
+                    keystone = keystoneclient(username=username,
+                                              password=password,
+                                              tenant_name=tenant_name,
+                                              auth_url=auth_url)
+                break
+            except KeyStoneException as e:
+                err = "Try nr {0}. Could not get keystone client, error: {1}"
+                logger.warning(err.format(i + 1, e))
+                time.sleep(5)
+        if not keystone:
+            raise
+        return keystone
+
+    def cleanup_network(self, networks_to_skip=[]):
```

Comment:
May be would be better to split this method to some parts, and call them here. Also, perhaps 'cleanup' or 'cleanup_env' will be a better name for this method.